static/app/views/dashboards/controls.tsx

```diff
@@ -178,6 +182,35 @@ function Controls({
                 {t('Export Dashboard')}
               </Button>
             </Feature>
+            <Feature features="dashboards-favourite">
+              <Button
+                size="sm"
+                aria-label={'dashboards-favourite'}
+                icon={
+                  <IconStar
+                    color={isFavorited ? 'yellow300' : 'subText'}
+                    isSolid={isFavorited}
+                    data-test-id={isFavorited ? 'yellow-star' : 'empty-star'}
+                  />
+                }
+                onClick={() => {
+                  setIsFavorited(!isFavorited);
+                  updateDashboardFavorite(
+                    api,
+                    organization.slug,
+                    dashboard.id,
+                    !isFavorited
+                  )
+                    .then(() => {
+                      dashboard.isFavorited = !dashboard.isFavorited;
+                    })
+                    .catch(() => {
+                      // If the api call fails, revert the state
+                      setIsFavorited(isFavorited);
+                    });
+                }}
+              />
```

Comment:
With async/await formatting, we're trying to use that instead of the promise format with `.then()` and `.catch()`. We could re-write this to follow that format

```suggestion
                onClick={async () => {
                  try {
                    setIsFavorited(!isFavorited);
                    await updateDashboardFavorite(
                      api,
                      organization.slug,
                      dashboard.id,
                      !isFavorited
                    );
                    dashboard.isFavorited = !dashboard.isFavorited;
                  } catch (error) {
                    // If the api call fails, revert the state
                    setIsFavorited(isFavorited);
                  }
                }}
```

Reply:
Also, `dashboard.isFavorited = !dashboard.isFavorited;` isn't updating the dashboard state properly, you would have to pass down a handler that calls the right function to set that state

Product question: If you duplicate a favourited dashboard, should that dashboard also be favourited?
I'm asking because I think that's the only flow I think that _would_ matter if we wanted to update the state with the favourited status. Otherwise, I think it's safe that we don't need to update the state for this modification